allennlp/training/trainer.py

```diff
@@ -117,10 +125,24 @@ def train(self) -> None:
                     loss = val_output_dict["loss"]
                     val_loss += loss.data.cpu().numpy()
 
-            # TODO(Mark): Add user specified metrics here, maybe a "metrics" key?
-            logger.info("Training Loss: %3f    Validation Loss: %3f ", train_loss, val_loss)
+                val_metrics = self._model.get_metrics(reset=True) or {}
+
+            message_template = "Training %s : %3f    Validation %s : %3f "
+            for name, value in metrics.items():
+                logger.info(message_template, name, value, name, val_metrics[name])
+            logger.info(message_template, "Loss", train_loss, "Loss", val_loss)
+
+            if self._validation_metric:
+                this_epoch = val_metrics[self._validation_metric]
+            else:
+                this_epoch = val_loss
+            if len(validation_metric_counts) > self._patience:
```

Comment:
This `counts` name is confusing to me - maybe `validation_metric_per_epoch`, or `epoch_validation_metrics`, or something?